mkosi/kmod.py

```diff
@@ -35,7 +35,7 @@ def filter_kernel_modules(
     if include:
         regex = re.compile("|".join(include))
         for m in modules:
-            rel = os.fspath(m.relative_to(modulesd / "kernel"))
+            rel = os.fspath(Path(*m.relative_to(modulesd).parts[2:]))
```

Comment:
Before, given some `'/usr/lib/modules/6.1.0-18-amd64/kernel/drivers/ptp/ptp_clockmatrix.ko'` this would produce `'drivers/ptp/ptp_clockmatrix.ko'` and now produces `'ptp/ptp_clockmatrix.ko'`. I don't think anybody will have matched the longer name, but in theory this could be a backwards incompatible change?

I assume naming collisions will not be a problem because things will be either in `modulesd` or in a subdirectory of `modulesd / "kernel"`, but is there a need to shorten `rel` at all instead of always using paths relative to `modulesd`?

Reply:
I think we should shorten to keep the same interface. But I did too much shortening, which is fixed in the latest push.